src/main/java/com/cinema/controller/MovieSessionController.java

```diff
@@ -0,0 +1,42 @@
+package com.cinema.controller;
+
+import com.cinema.model.dto.MovieSessionRequestDto;
+import com.cinema.model.dto.MovieSessionResponseDto;
+import com.cinema.model.dto.mapper.MovieSessionDtoMapper;
+import com.cinema.service.MovieSessionService;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionDtoMapper movieSessionDtoMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionDtoMapper movieSessionDtoMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+    }
+
+    @PostMapping
+    public void addMovieSession(@RequestBody MovieSessionRequestDto session) {
+        movieSessionService.add(movieSessionDtoMapper.createSessionRequest(session));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSession(
+            @RequestParam Long id, @DateTimeFormat(pattern = "yyyy-MM-dd") LocalDate date) {
```

Comment:
```suggestion
            @RequestParam Long id, @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date) {
```
или я б вынес "yyyy-MM-dd" в константу

Reply:
DateTimeFormat.ISO.DATE чудова ідея, дякую

Reply:
let's use DateTimeFormat.ISO.DATE